packages/svelte/src/compiler/compile/Component.js

```diff
@@ -772,12 +772,23 @@ export default class Component {
 			}
 			const writable =
 				node.type === 'VariableDeclaration' && (node.kind === 'var' || node.kind === 'let');
-			const imported = node.type.startsWith('Import');
+
+			let immutable = false;
+			if (node.type === 'VariableDeclaration' && node.kind === 'const') {
+				immutable = true;
+				for (const declaration of node.declarations) {
+					if (declaration.init.type !== 'Literal' || typeof declaration.init.value === 'object') {
```

Comment:
When will the typeof declaration.init.value  will be and object?

Reply:
If we have something like `const x = { prop1: 'foo', prop2: 'bar' };` then it'd be an object. In that case the variable is not immutable because the properties of the variable might be mutated. Potentially in the future we could detect it's immutable if there's a call to `Object.freeze` or a `/** @readonly */` JSDoc, but for now I think that's too complicated an optimization and not necessary.

Reply:
Ah, yes. You are right. Updated